src/pit/viper/config.py

```diff
@@ -0,0 +1,142 @@
+"""Load config from file."""
+from pathlib import Path
+from typing import Any
+
+from pit.viper import _io
+
+_config_path: Path | None = None
+_config_name: str = ""
+_config_type: str = ""
+_config: dict[str, Any] = {}
+
+
+def set_config_name(name: str) -> None:
+    """Set the config name.
+
+    Parameters
+    ----------
+    name : str
+        Name of the config file.
+    """
+    global _config_name  # noqa: PLW0603
+    _config_name = name
+
+
+def set_config_path(path: Path) -> None:
+    """Set the config path.
+
+    Parameters
+    ----------
+    path : Path
+        Path to the config file or the directory containing the config
+        file.
+    """
+    global _config_path  # noqa: PLW0603
+    _config_path = path
+
+
+def set_config_type(type_: str) -> None:
+    """Set the config type.
+
+    Supported config formats: json, toml, yaml, yml.
+
+    Parameters
+    ----------
+    type_ : str
+        Config format.
+    """
+    global _config_type  # noqa: PLW0603
+    _config_type = type_.lower()
+
+
+def getconf(key: str, default: Any = None) -> Any:
+    """Get a value from the config.
+
+    Parameters
+    ----------
+    key : str
+        Key to get the value for.
+    default : Any, optional
+        Default value to return if the key is not found, by default None
+
+    Returns
+    -------
+    Any
+        Value for the key.
+    """
+    keys = key.split(".")
+    value = _config
+
+    for k in keys:
+        if isinstance(value, dict):
```

Comment:
If this is False, surely the default is not used?!

Reply:
This only works since we are certain that `_config` is a `dict`. Note that even if the `key` was an empty string, `keys` would be a list only containing one entry (i.e., the empty string). Therefore, we would also return the default in that case.